frappe/email/doctype/contact/contact.py

```diff
@@ -135,3 +135,13 @@ def contact_query(doctype, txt, searchfield, start, page_len, filters):
 			'link_doctype': filters.get('link_doctype'),
 			'link_name': filters.get('link_name')
 		})
+
+def contact_links(doctype, txt, searchfield, start, page_len, filters):
+	if not txt: txt = ""
+	txt = txt.lower()
+	return [[d] for d in get_link_doctypes() if txt in d.lower()]
+
+@frappe.whitelist()
+def get_link_doctypes():
+	return [x.parent for x in frappe.db.get_values("DocField", {"fieldname": 'contact_html'}, "parent", order_by="name", as_dict=1)]
```

Comment:
oh right should filter the text with sql but customer and supplier and sales partner arent link fields they are one to many this is my solution to not hard coding which probably would b more ideal as would allow the addition of both lead's and users